sdk/iOS/src/MSSyncContextReadResult.h

```diff
@@ -13,7 +13,7 @@
 /// @{
 
 /// Creates a MSSyncContextReadResult with the given items and total count.
-- initWithCount:(NSInteger)count items:(NSArray *)items;
+- (nonnull instancetype)initWithCount:(NSInteger)count items:(nonnull NSArray *)items;
```

Comment:
The only use I found was here: https://github.com/Azure/azure-mobile-services/blob/MobileApp/sdk/iOS/src/MSCoreDataStore.m#L201-L306

For this, the results array is allocated and assigned inside the performBlockAndWait. Hence I assumed it would be nonnull and just empty if there were zero results.